Add shadowrootslotassignment attribute (#43753)
mdn/content: files/en-us/web/html/reference/elements/slot/index.md

sidebar: htmlsidebar
---

The **`<slot>`** [HTML](/en-US/docs/Web/HTML) element is a placeholder inside a [Web Component](/en-US/docs/Web/API/Web_components) that you can fill with your own markup when the component is used.
This lets you create separate DOM trees and present them together.

> A slot can contain default content, such as plain text, other HTML elements, or other web components.

Reviewer: Can the slot also be filled with plain text, other HTML elements, or other web components? If so, and I'd imagine so, it seems a bit weird to comment that this is what the default content can be, as if it can't also be filled with those content types.

Better to say something like this, maybe:

> Slots can contain plain text, other HTML elements, or other web components. A slot can also contain default content, which is displayed if the slot is not assigned other content when the web component is used.

Reviewer 2: Thanks!!

```suggestion
Slots can contain plain text, other HTML elements, or other web components. A slot can also contain default content, which is displayed if the slot is not assigned other content when the web component is used.
```

> This is displayed if the slot is not assigned another element when the web component is used.

Reviewer 2: ```suggestion
```

## Attributes

This element includes the [global attributes](/en-US/docs/Web/HTML/Reference/Global_attributes).

- `name`
  - : The slot's name.

>     A _named slot_ is a `<slot>` element with a `name` attribute, while unnamed slots have no `name` attribute, and the name defaults to the empty string.

Reviewer: ```suggestion
    A _named slot_ is a `<slot>` element with a `name` attribute, while an _unnamed slot_ has no `name` attribute, and the name defaults to the empty string.
```

>     When the associated shadow root uses [named slot assignment](/en-US/docs/Web/HTML/Reference/Elements/template#named), the slot's containing element is rendered with elements in the same host that have a [`slot` attribute](/en-US/docs/Web/API/Element/slot) with the same name.

Reviewer: "the slot's containing element is rendered with elements" is a bit unclear — does this mean that on rendering, a named `<slot>` element is replaced by an element in the same host that has a `slot` attribute set equal to the same name?

Reviewer 2: Yes. Thanks. Fixed in https://github.com/mdn/content/pull/43753/changes/1ff77a470e0f7ba917b724455feab2b6092342a3

    All elements that don't have a `slot` attribute are rendered in the first unnamed `<slot>` element, which is referred to as the _default slot_.

>     Slot names should be unique per shadow root: if you have two slots with the same name, all of the elements with a matching `slot` attribute will be assigned to the first slot with that name.

Reviewer: In such a situation, do both elements get rendered, or just the last one in the source order?

Reviewer 2: They all get rendered in the first slot with the matching name. If they weren't to be rendered they would not be assigned - they'd be dropped.

I hope this makes it more clear.

```suggestion
    Slot names should be unique per shadow root: if you have two slots with the same name, _all_ of the elements with a matching `slot` attribute are rendered in the first slot that has the name.
```
